src/ssh-connection-manager.ts

```diff
@@ -68,29 +73,44 @@ export class SshConnectionManager {
   }
 
   async getSftp(address: string): Promise<Sftp | undefined> {
+    const username = getUsername(this.configManager, address);
     try {
-      const conn = await this.getConnection(address);
+      const conn = await this.getConnection(address, username);
       return new Sftp(conn);
     } catch (err) {
       if (this.isAuthenticationError(err)) {
-        const username = await this.promptForUsername(address);
-
-        if (username) {
+        this.displayAuthenticationError(err.level, username, address);
+        if (await this.promptForUsername(address)) {
           return await this.getSftp(address);
         }
-
         this.showUsernameRequiredError();
```

Comment:
Since we're displaying the Authentication error, I think we can omit `showUsernameRequiredError`